### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by AsumVictor**
# Milestone 7: Testing
In this PR, we have added testing to the components

 ## Type of change
 * [ ] Added react testing library
 * [ ] Created a Test folder in the component directory
 * [ ] Added test for components

 
 # Checklist:
 * [ ]  Our code follows the style guidelines of this project
 * [ ] We have performed a self-review of my code
 * [ ] There is no linter error in my code
 * [ ]  Our changes generate no new warnings

### CODE REVIEW

File: src/components/Test/Quote.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    );
+
+    const quoteElement = getByText(quoteText);
+    const authorElement = getByText(authorName);
+    const avatarElement = getByAltText(authorName);
+
+    expect(quoteElement).toBeInTheDocument();
+    expect(authorElement).toBeInTheDocument();
+    expect(avatarElement).toBeInTheDocument();
+  });
```
Comment (thecodechaser) [inline src/components/Test/Quote.test.js:21]: - [ ] Some of the tests are failing, Please make sure to fix them and all tests should pass without any errors.

image

File: src/components/Test/Quote.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import '@testing-library/jest-dom/extend-expect';
+import Quote from '../Quote';
+
+describe('Quote component', () => {
+  it('should render quote and author correctly', () => {
+    const quoteText = 'This is a test quote';
+    const authorName = 'Test Author';
+    const { getByText, getByAltText } = render(
+      <Quote quote={quoteText} author={authorName} />,
+    );
```
Comment (thecodechaser) [inline src/components/Test/Quote.test.js:12]: - [x] Please make sure to write tests for the `Calculator` and `Header` components as requested in the project requirements to write tests for all the components.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @AsumVictor, and @Dorcas126👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [ ] I would suggest adding steps in the README file to run the tests, It will make the README file more professional. Example: `npm run test`.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Test/Quote.test.js (hunk 1 above)
Comment (AsumVictor) [reply src/components/Test/Quote.test.js:21]: Thank you very much. we have added more test case and all has ticked passed

File: src/components/Test/Quote.test.js (hunk 2 above)
Comment (AsumVictor) [reply src/components/Test/Quote.test.js:12]: We have resolved this. Thank you very much!

**Review by roshan-bajgain — APPROVED** [review]
Hi @AsumVictor and @Dorcas126, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- All test are passing ✔️ 

## Optional suggestions
- [ ] Please follow the optional suggestion given by the previous reviewer.
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AsumVictor** [comment]
Thank you very much!

**Pull request merged**